example/extensions/lib_custom_op/README.md

```diff
@@ -0,0 +1,118 @@
+                std::string> attrs,
+                int* num_in,
+                int* num_out)
+
+
+* [inferType](./gemm_lib.cc#L124) - Type Inference:
+    * This function specifies how custom operator infers output data types using input data types.
+
+            MXReturnValue inferType(
+                std::map<std::string, std::string> attrs,
+                std::vector<int> &intypes,
+                std::vector<int> &outtypes)
+
+* [inferShape](./gemm_lib.cc#L143) - Shape Inference:
+    * This function specifies how custom operator infers output tensor shape using input shape.
+
+            MXReturnValue inferShape(
+                std::map<std::string, std::string> attrs,
+                std::vector<std::vector<unsigned int>> &inshapes,
+                std::vector<std::vector<unsigned int>> &outshapes)
+
+* [forward](./gemm_lib.cc#L56) - Forward function:
+    * This function specifies the computation of forward pass of the operator.
+
+            MXReturnValue forward(
+                std::map<std::string, std::string> attrs,
+                std::vector<MXTensor> inputs,
+                std::vector<MXTensor> outputs,
+                OpResource res)
+
+* [REGISTER_OP(my_op_name) Macro](./gemm_lib.cc#L169):
+    * This macro registers custom operator to all MXNet APIs by its name, and you need to call setters to bind the above functions to the registered operator.
+
+            REGISTER_OP(my_op_name)
+            .setForward(forward)
+            .setParseAttrs(parseAttrs)
+            .setInferType(inferType)
+            .setInferShape(inferShape);
+
+Also there are some optional functions you can specify:
+
+* [backward](./gemm_lib.cc#L90) - Backward Gradient function:
+    * This function specifies the computation of backward pass of the operator.
+
+            MXReturnValue backward(
+                std::map<std::string, std::string> attrs,
+                std::vector<MXTensor> inputs,
+                std::vector<MXTensor> outputs,
+                OpResource res)
+
+* [mutateInputs](./gemm_lib.cc#L214) - Specify mutable input:
+    * This function allows you to mark some inputs to be mutable inputs, useful when using aux parameters for BatchNorm-like operators.
+
+            MXReturnValue mutateInputs(
+                std::map<std::string, std::string> attrs,
+                std::vector<int> &input_indices)
+
+Let’s take a closer look at those registry functions:
+
+* **parseAttrs**: This function takes 3 arguments. 1st argument is an input, which is the attributes passed all the way from Python code. When user calls `mx.nd.my_op_name(s,t,keyword=1)`, the keyword is passed to the attributes as an entry of the map. 2nd & 3rd arguments are outputs, and you need to set number of inputs and outputs values to those placeholders.  If the number of input and output tensors are fixed, you can use hard-coded numbers. Otherwise you can get the user-specified attributes to determine the number of inputs and outputs.
```

Comment:
```suggestion
* **parseAttrs**: This function takes three arguments. The 1st argument is an input, which is the attributes passed all the way from Python code. When you call `mx.nd.my_op_name(s,t,keyword=1)`, the keyword is passed to the attributes as an entry of the map. The 2nd & 3rd arguments are outputs, and you need to set number of inputs and outputs values to those placeholders.  If the number of input and output tensors are fixed, you can use hard-coded numbers. Otherwise, you can get the user-specified attributes to determine the number of inputs and outputs.
```